pinot-core/src/test/java/com/linkedin/pinot/queries/HllIndexSentinelTest.java

```diff
@@ -0,0 +1,183 @@
+ * In minor places, the logic is not the same as QueriesSentinelTest.
+ * E.g. different data loaded, pre-assigned query columns, etc.
+ */
+public class HllIndexSentinelTest {
+  private static final Logger LOGGER = LoggerFactory.getLogger(HllIndexSentinelTest.class);
+
+  private static final String timeColumnName = "daysSinceEpoch";
+  private static final TimeUnit timeUnit = TimeUnit.DAYS;
+  private static final String hllDeriveColumnSuffix = HllConstants.DEFAULT_HLL_DERIVE_COLUMN_SUFFIX;
+  private static final int hllLog2m = HllConstants.DEFAULT_LOG2M;
+  private static final double approximationThreshold = 0.001;
+  private static final String tableName = "testTable";
+
+  private static QueryExecutor QUERY_EXECUTOR;
+  private static FileBasedInstanceDataManager instanceDataManager;
+  private static PropertiesConfiguration serverConf;
+  private SegmentWithHllIndexCreateHelper helper;
+  private String segmentName;
+
+  private static final String AVRO_DATA = "data/test_data-sv.avro";
+
+  private static final Set<String> columnsToDeriveHllFields =
+      new HashSet<>(Arrays.asList("column1", "column2", "column3",
+          "count", "weeksSinceEpochSunday", "daysSinceEpoch",
+          "column17", "column18"));
+
+  private static final HllConfig hllConfig =
+      new HllConfig(columnsToDeriveHllFields, hllLog2m, hllDeriveColumnSuffix);
+
+  private void setupTableManager() throws Exception {
+    TableDataManagerProvider.setServerMetrics(new ServerMetrics(new MetricsRegistry()));
+    serverConf =  new TestingServerPropertiesBuilder(tableName).build();
+    serverConf.setDelimiterParsingDisabled(false);
+
+    instanceDataManager = FileBasedInstanceDataManager.getInstanceDataManager();
+    instanceDataManager.init(new FileBasedInstanceDataManagerConfig(serverConf.subset("pinot.server.instance")));
+    instanceDataManager.start();
+  }
+
+  @BeforeClass
+  public void setup() throws Exception {
+    setupTableManager();
+
+    // Setup Segment
+    helper = new SegmentWithHllIndexCreateHelper(tableName, AVRO_DATA, timeColumnName, timeUnit);
+    SegmentIndexCreationDriver driver = helper.build(true, hllConfig);
+    File segmentFile = helper.getSegmentDirectory();
+    segmentName = helper.getSegmentName();
+    LOGGER.debug("************************** Segment Directory: " + segmentFile.getAbsolutePath());
+
+    // Load Segment
+    final IndexSegment indexSegment = ColumnarSegmentLoader.load(segmentFile, ReadMode.heap);
+    instanceDataManager.getTableDataManager(tableName).addSegment(indexSegment);
+
+    // Init Query Executor
+    QUERY_EXECUTOR = new ServerQueryExecutorV1Impl(false);
+    QUERY_EXECUTOR.init(serverConf.subset("pinot.server.query.executor"), instanceDataManager, new ServerMetrics(
+        new MetricsRegistry()));
+  }
+
+  @AfterClass
+  public void tearDown() {
+    helper.cleanTempDir();
+  }
+
+  @Test
+  public void testFastHllNoGroupBy() throws Exception {
```

Comment:
Add a test that executes query for multiple segments. You can do that as a separate PR